stripe/src/main/java/com/stripe/android/model/PaymentIntent.java

```diff
@@ -0,0 +1,299 @@
+    static final String FIELD_CLIENT_SECRET = "client_secret";
+    static final String FIELD_CONFIRMATION_METHOD = "confirmation_method";
+    static final String FIELD_CURRENCY = "currency";
+    static final String FIELD_DESCRIPTION = "description";
+    static final String FIELD_LIVEMODE = "livemode";
+    static final String FIELD_NEXT_SOURCE_ACTION = "next_source_action";
+    static final String FIELD_RECEIPT_EMAIL = "receipt_email";
+    static final String FIELD_SOURCE = "source";
+    static final String FIELD_STATUS = "status";
+
+    private String mId;
+    private String mObjectType;
+    private List<String> mAllowedSourceTypes;
+    private Long mAmount;
+    private Long mCanceledAt;
+    private String mCaptureMethod;
+    private String mClientSecret;
+    private String mConfirmationMethod;
+    private Long mCreated;
+    private String mCurrency;
+    private String mDescription;
+    private Boolean mLiveMode;
+    private Map<String, Object> mNextSourceAction;
+    private String mReceiptEmail;
+    private String mSource;
+    private String mStatus;
+
+    @NonNull
+    public String getId() {
+        return mId;
+    }
+
+    @NonNull
+    public List<String> getAllowedSourceTypes() {
+        return mAllowedSourceTypes;
+    }
+
+    @NonNull
+    public Long getAmount() {
+        return mAmount;
+    }
+
+    @Nullable
+    public Long getCanceledAt() {
+        return mCanceledAt;
+    }
+
+    @Nullable
+    public String getCaptureMethod() {
+        return mCaptureMethod;
+    }
+
+    @NonNull
+    public String getClientSecret() {
+        return mClientSecret;
+    }
+
+    @Nullable
+    public String getConfirmationMethod() {
+        return mConfirmationMethod;
+    }
+
+    @Nullable
+    public Long getCreated() {
+        return mCreated;
+    }
+
+    @NonNull
+    public String getCurrency() {
+        return mCurrency;
+    }
+
+    @Nullable
+    public String getDescription() {
+        return mDescription;
+    }
+
+    @NonNull
+    public Boolean isLiveMode() {
+        return mLiveMode;
+    }
+
+    @Nullable
+    public Map<String, Object> getNextSourceAction() {
+        return mNextSourceAction;
+    }
+
+    public Uri getAuthorizationUrl() {
+        if ("requires_source_action".equals(mStatus) &&
+                mNextSourceAction.containsKey("type") &&
+                "authorize_with_url".equals(mNextSourceAction.get("type")) &&
+                mNextSourceAction.containsKey("value") &&
+                mNextSourceAction.get("value") instanceof Map) {
+            Map<String, String> urlMap = (Map) mNextSourceAction.get("value");
+            if (urlMap.containsKey("url")) {
+                return Uri.parse(urlMap.get("url"));
+            }
+        }
+        return null;
+    }
+
+    public String getReceiptEmail() {
+        return mReceiptEmail;
+    }
+
+    public String getSource() {
+        return mSource;
+    }
+
+    public String getStatus() {
```

Comment:
Missing nullability annotations from `getAuthorizationUrl`, `getReceiptEmail`, `getSource`, and `getStatus`. I don't know what the default is, but I suspect you want status to be non-null and the others to be nullable.